docs/key-issues/multiple_architectures.md

```diff
@@ -0,0 +1,266 @@
+# Platforms with multiple CPU architectures
+
+One important subset of ABI concerns is the CPU architecture for which a binary
+artefact has been built. Attempting to run a binary on hardware that doesn't
+match the CPU architecture (or architecture variant[^1]) for which the binary
+was built will generally lead to crashes, even if the ABI being used is
+otherwise compatible.
+
+[^1]:
+    E.g., the x86-64 architecture has a range of well-known extensions, such as
+    SSE, SSE2, SSE3, AVX, AVX2, AVX512, etc.
+
+## Current state
+
+Historically, it could be assumed that an executable or library would be
+compiled for a single CPU architecture. On the rare occasion that an operating
+system was available for mulitple CPU architectures, it became the
+responsibility of the user to find (or compile) a binary that was compiled for
+their host CPU architecture.
+
+However, we now see operating system platforms where multiple CPU architectures
+are supported:
+
+* In the early days of Windows NT, both x86 and DEC Alpha CPUs were supported
+* Windows 10 supports x86, x86-64, ARMv7 and ARM64; Windows 11 supports x86-64
+  and ARM64.
+* Due to its open source nature, Linux tends to support all CPU architectures for
+  which someone is interested enough to author & provide support in the kernel,
+  see [here](https://en.wikipedia.org/wiki/List_of_Linux-supported_computer_architectures).
+* Apple transitioned Mac hardware from PowerPC to Intel (x86-64) CPUs, providing
+  a forwards compatibility path for binaries
+* Apple is currently transitioning Mac hardware from Intel (x86-64) to
+  Apple Silicon (ARM64) CPUs, again providing a forwards compatibility
+  path
+* Apple supports ARMv6, ARMv7, ARMv7s, ARM64 and ARM64e on iOS
+* Android currently supports ARMv7, ARM64, x86, and x86-64; it has historically
+  also supported ARMv5 and MIPS
+
+CPU architecture compatibility is a necessary, but not sufficient criterion for
+determining binary compatibility. Even if two binaries are compiled for the same
+CPU architecture, that doesn't guarantee [ABI compatibility](abi.md).
+
+In some respects, CPU architecture compatibility could be considered a superset
```

Comment:
I'm not sure that this paragraph, and this sentence in particular, helps. I suspect users may think "really, how?" and there is no answer as of today because there are no solutions for GPU support at all yet, and it's unclear where that will go in the future.

Reply:
It was more of an idle thought than a particularly deep insight, so if you think it's more confusing than help, I'll delete it.